cli/azd/pkg/tools/github/github.go

```diff
@@ -33,10 +48,75 @@ type GitHubCli interface {
 	GitHubActionsExists(ctx context.Context, repoSlug string) (bool, error)
 }
 
-func NewGitHubCli(commandRunner exec.CommandRunner) GitHubCli {
-	return &ghCli{
+func NewGitHubCli(ctx context.Context, console input.Console, commandRunner exec.CommandRunner) (GitHubCli, error) {
+	return newGitHubCliImplementation(ctx, console, commandRunner, http.DefaultClient, downloadGh, extractGhCli)
+}
+
+// cGitHubCliVersion is the minimum version of GitHub cli that we require (and the one we fetch when we fetch bicep on
+// behalf of a user).
+var cGitHubCliVersion semver.Version = semver.MustParse("2.22.1")
+
+// newGitHubCliImplementation is like NewGitHubCli but allows providing a custom transport to use when downloading the
+// GitHub CLI, for testing purposes.
+func newGitHubCliImplementation(
+	ctx context.Context,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+	transporter policy.Transporter,
+	acquireGitHubCliImpl GetGitHubCliImplementation,
+	extractImplementation ExtractGitHubCliFromFileImplementation,
+) (GitHubCli, error) {
+	if override := os.Getenv("AZD_GH_CLI_TOOL_PATH"); override != "" {
+		log.Printf("using external github cli tool: %s", override)
+
+		return &ghCli{
+			path:          override,
+			commandRunner: commandRunner,
+		}, nil
+	}
+
+	githubCliPath, err := azdGithubCliPath()
+	if err != nil {
+		return nil, fmt.Errorf("finding github cli: %w", err)
+	}
+
+	if _, err = os.Stat(githubCliPath); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, fmt.Errorf("finding github cli: %w", err)
+	}
+	if errors.Is(err, os.ErrNotExist) {
```

Comment:
Should we look for `gh` in `PATH`? I feel like when @ellismg had implemented `bicep` installation, one aspect that was different than is that it's unlikely for users to use a standalone `bicep` install, because `bicep` is usually bundled with `az`. 

Here, it's different and my preference would be to respect and play nice with the existing installation.

Reply:
co-existing with some existing `gh cli` brings more issues than benefits. 
The `gh cli` binary is `39MB` in linux and `34MB` in Windows, it's really small enough to just get and let it handle authentication for us.

For users who has one version installed, we don't want to worry about checking if their version is compatible. That's why we pin our own version.

For user who doesn't have the gh cli installed, I don't want them to feel like we are installing something in their PC. I want this small binary to be considered as part of `azd`.  We can even consider about removing the binary once `azd pipeline config` finish (but I don't think is worth it for a few MB)

Reply:
I'm okay with pinning and our own copy.